1.9/networking/service-and-pod-networking.md

```diff
@@ -0,0 +1,422 @@
+
+Each DC/OS service is given a single port by default, so `$PORT0` will normally be available, except for services that specifically declare "no ports".
+
+Variables are generated for all services, regardless of whether the service uses the [UCR or Docker containerizer](/docs/1.9/deploying-services/containerizers/).
+
+It is **highly recommended** to name the ports of a service to provide clarity with respect to the service configuration and intended use of each port.
+
+When using `container` or `container/bridge` mode networking, you must bind your application to the `containerPort`s you have specified in your `portMapping`s.
+
+If you have set `containerPort` to `0`, this will be the same as `hostPort` and you can use the `$PORTxxx` environment variables.
+
+## Discovery Via Mesos
+
+### `DiscoveryInfo` and port `labels`
+
+* `labels` may be defined for items of `portDefinitions` as well as for items of `portMappings`. These labels are sent to Mesos via `DiscoveryInfo` protobufs at instance-launch time.
+* Given a mapping or endpoint, Marathon generates a port `DiscoveryInfo` for every combination of specified protocols and associated networks. For instance, if an `Endpoint` specifies 2 protocols and is associated with 3 container networks, then a total of 6 `DiscoveryInfo` protobufs would be generated for that single `Endpoint`.
+* Marathon injects a `network-scope` label into the port `DiscoveryInfo` to disambiguate between a *host-port* and *container-port*.
+    * A scope value of `host` is used for *host-port* discovery.
+    * A scope value of `container` is used for *container-port* discovery.
+* For *container-port* discovery, Marathon also injects a `network-name` label into the respective port `DiscoveryInfo`.
+
+### Virtual addresses
+
+See the documentation for [virtual addresses (VIPs)](/docs/1.9/networking/load-balancing-vips/virtual-ip-addresses/).
+
+# Examples
+
+## `host` Mode
+
+`host` mode networking is the default networking mode for all services. If your service uses Docker containers, it not necessary to `EXPOSE` ports in your `Dockerfile`.
+
+### Using `host` Mode
+
+Host mode is enabled by default for all services and all container types.
+If you wish to be explicit, you can also specify it manually in the `networks` property:
+
+```json
+  "networks": [ { "mode": "host" } ],
+  "container": {
+    "type": "MESOS",
+    "docker": {
+      "image": "my-image:1.0"
+    }
+  },
+```
+
+### Specifying Ports
+
+You can specify the ports that are available through the `portDefinitions` array:
+
+```json
+    "portDefinitions": [
+      {"port": 0, "name": "http"}, {"port": 0, "name": "https"}, {"port": 0, "name": "mon"}
+    ],
+```
+
+In this example, we specify three dynamically assigned host ports, which would then be available to our command via the environment variables `$PORT_HTTP`, `$PORT_HTTPS` and `$PORT_MON`.
+
```

Comment:
What's meant by "command" here? Is it that they're available to the app/service?